### ISSUE TITLE
add conda deactivate --all flag

### ISSUE THREAD

**Issue by tutufan** [conda/conda#7609]
It would be useful to have a `--all` flag to `conda deactivate` that would have the effect of deactivating _all_ activated environments.

This can currently be done in a loop, but it's not efficient, and in some environments such looping is not available.

(see also https://github.com/conda/conda/issues/7407)

**Comment by mingwandroid** [comment]
While I support this feature I'd caution people strongly against stacking environments.

Stacking envs works against isolation and reproducibility or at least complicates them greatly.

**Comment by tutufan** [comment]
@mingwandroid Yeah, not crazy about it either.  But as long as the feature exists, from an admin perspective, we have to deal with it.

**Comment by hmaarrfk** [comment]
@ mingwandroid is there a way that I can automatically `deactivate` when i call `activate`? I often activate an environment while I am in base and it causes lots of confusion.....

**Comment by kalefranz** [comment]
We’re releasing conda 4.6.0 beta0 to canary on a week or so. It has several improvements in this regard.

**Comment by hmaarrfk** [comment]
@kalefranz if that  was in response to a comment I made, thanks for the answer. I removed my comment because I didn't want to hijack the thread. For reference my question was : 

> Is there any way to automatically `deactivate` environments if we call `activate`? I often activate environment from within `base` which causes confusion.

**Comment by michaelkarlcoleman** [comment]
Ha.  Bots are one thing.  Passive-aggressive bots are a bit more, though.  Cheers, Mr. @github-actions !

**Comment by michaelkarlcoleman** [comment]
@jezdez My apologies--I meant that mostly tongue-in-cheek.  It _is_ mildly annoying to have bug reports being closed (or proceeding there) without resolution, but then, I don't have a project with 1700 open bug reports.  :-)  Thank you for your efforts!
